cmake/Modules/FindComputeCpp.cmake

```diff
@@ -370,23 +414,48 @@ endfunction()
 #  binaryDir : Intermediate directory to output the integration header.
 #  sourceFiles : Source files to be compiled for SYCL.
 #
-function(add_sycl_to_target targetName binaryDir sourceFiles)
+function(add_sycl_to_target)
+  set(options)
+  set(oneValueArgs TARGET BINARY_DIR)
+  set(multiValueArgs SOURCES)
+  cmake_parse_arguments(add_sycl_to_target "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+  set(targetName ${add_sycl_to_target_TARGET})
+  set(binaryDir ${add_sycl_to_target_BINARY_DIR})
+  set(sourceFiles ${add_sycl_to_target_SOURCES})
 
-  set(sourceFiles ${sourceFiles} ${ARGN})
   set(fileCounter 0)
   target_include_directories(
     ${targetName} SYSTEM
     PRIVATE ${OpenCL_INCLUDE_DIR}
     PRIVATE ${COMPUTECPP_INCLUDE_DIRECTORY}
   )
-  # Add custom target to run compute++ and generate the integration header
-  foreach(sourceFile ${sourceFiles})
-    if(NOT IS_ABSOLUTE ${sourceFile})
-      set(sourceFile "${CMAKE_CURRENT_SOURCE_DIR}/${sourceFile}")
+
+  if(COMPUTECPP_USE_COMPILER_DRIVER AND isCompilerDriverSupported)
+    # Convert argument list format
+    separate_arguments(COMPUTECPP_DEVICE_COMPILER_FLAGS)
+    # Set all flags required to compile the target when using the compiler driver
+    set(computeCppCompileOptions ${COMPUTECPP_DEVICE_COMPILER_FLAGS})
+    get_property(includeAfter TARGET ${targetName} PROPERTY COMPUTECPP_INCLUDE_AFTER)
+    if(includeAfter)
+      list(APPEND computeCppCompileOptions -fsycl-ih-last)
     endif()
-    __build_spir(${targetName} ${sourceFile} ${binaryDir} ${fileCounter})
-    MATH(EXPR fileCounter "${fileCounter} + 1")
-  endforeach()
+    target_compile_options(${targetName} PUBLIC ${computeCppCompileOptions})
+  else()
+    if(NOT binaryDir)
+      message(WARNING " ComputeCpp Warning: BINARY_DIR was not specified. Using CMAKE_CUURENT_BINARY_DIR which is \"${CMAKE_CUURENT_BINARY_DIR}\"")
+      set(binaryDir ${CMAKE_CURRENT_BINARY_DIR})
+    endif()
+    if(NOT sourceFiles)
+      message(FALTAL_ERROR " ComputeCpp Error: SOURCES was not specified. At least one source file must be provided.")
```

Comment:
Is this really an error? I can think of a use case where a user wants to link against ComputeCPP but doesn't actually need to compile any integration headers. One way of doing that would be to call `add_sycl_to_target` without specifying any source files (and so also not needing to specify a binary dir).

You would need to add a check before the `foreach` below though.

If you do decide to keep this error, fix the typo `FALTAL_ERROR` -> `FATAL_ERROR`.